cpp/src/traversal/sssp.cu

```diff
@@ -314,91 +310,42 @@ void sssp(Graph* graph,
     // double
 
-    void* d_edge_data;
-    graph->adjList->edge_data = new gdf_column;
     cudaStream_t stream{nullptr};
-
-    // If distances array is given and is double, generate the weights in
-    // double
-    if (distances && typeid(WT) == typeid(double)) {
-      std::vector<double> h_edge_data(graph->adjList->indices->size, 1.0);
-      size_t edge_data_size = sizeof(double) * h_edge_data.size();
-      ALLOC_TRY((void**)&d_edge_data, edge_data_size, stream);
-      CUDA_TRY(cudaMemcpy(d_edge_data,
-                          &h_edge_data[0],
-                          edge_data_size,
-                          cudaMemcpyHostToDevice));
-      gdf_column_view(graph->adjList->edge_data,
-                      d_edge_data,
-                      nullptr,
-                      graph->adjList->indices->size,
-                      GDF_FLOAT64);
-
-    } else {
-      // Else generate float
-      std::vector<float> h_edge_data(graph->adjList->indices->size, 1.0);
-      size_t edge_data_size = sizeof(float) * h_edge_data.size();
-      ALLOC_TRY((void**)&d_edge_data, edge_data_size, stream);
-      CUDA_TRY(cudaMemcpy(d_edge_data,
-                          &h_edge_data[0],
-                          edge_data_size,
-                          cudaMemcpyHostToDevice));
-      gdf_column_view(graph->adjList->edge_data,
-                      d_edge_data,
-                      nullptr,
-                      graph->adjList->indices->size,
-                      GDF_FLOAT32);
-    }
+    std::vector<WT> h_edge_data(graph.number_of_edges, static_cast<WT>(1));
+    size_t edge_data_size = sizeof(WT) * h_edge_data.size();
+    ALLOC_TRY((WT**)&edge_weights_ptr, edge_data_size, stream);
+    CUDA_TRY(cudaMemcpy(edge_weights_ptr,
+                        &h_edge_data[0],
+                        edge_data_size,
+                        cudaMemcpyHostToDevice));
+    should_free_weights = true;
   } else {
-    // Got weighted graph
-    CUGRAPH_EXPECTS(
-        graph->adjList->edge_data->size == graph->adjList->indices->size,
-        "Graph sizes mismatch");
-    // TODO fix me after gdf_column is removed from Graph
-    CUGRAPH_EXPECTS(graph->adjList->edge_data->dtype == GDF_FLOAT32 ||
-                    graph->adjList->edge_data->dtype == GDF_FLOAT64,
-                "Invalid API parameter");
-    // TODO fix me after gdf_column is removed from Graph
-    // if (distances) CUGRAPH_EXPECTS(typeid(distances) == typeid(graph->adjList->edge_data), "distances and weights type mismatch");
-
     // SSSP is not defined for graphs with negative weight cycles
     // Warn user about any negative edges
-    if (graph->prop && graph->prop->has_negative_edges == GDF_PROP_TRUE)
+    if (graph.prop.has_negative_edges == experimental::PropType::PROP_TRUE)
       std::cerr << "WARN: The graph has negative weight edges. SSSP will not "
                    "converge if the graph has negative weight cycles\n";
+    edge_weights_ptr = (WT*)graph.edge_data;
```

Comment:
Why do we cast to `WT*` here?
We are pretty sure `graph.edge_data` is `WT*` at this point otherwise it would be detected at compile time it seems.

Reply:
If it is about the const, this may go away in https://github.com/rapidsai/cugraph/pull/780 (also being discussed there)

Reply:
Udpated in https://github.com/rapidsai/cugraph/pull/764/commits/1528aebe9ea39d6ac0060b1068a76f8c2a36056c: SSSP and BFS now use const pointers